lib/bloc/fee_options/fee_options_bloc.dart

```diff
@@ -17,56 +17,78 @@ class FeeOptionsBloc extends Cubit<FeeOptionsState> {
   FeeOptionsBloc(this._breezSDK) : super(FeeOptionsState.initial());
 
   /// Fetches the current recommended fees
-  Future<List<FeeOption>> fetchFeeOptions(String toAddress) async {
+  Future<List<FeeOption>> fetchFeeOptions({required String toAddress, String? swapAddress}) async {
     RecommendedFees recommendedFees;
     try {
       recommendedFees = await _breezSDK.recommendedFees();
       _log.info(
         "fetchFeeOptions recommendedFees:\nfastestFee: ${recommendedFees.fastestFee},"
         "\nhalfHourFee: ${recommendedFees.halfHourFee},\nhourFee: ${recommendedFees.hourFee}.",
       );
-      return await _constructFeeOptionList(toAddress, recommendedFees);
+      return await _constructFeeOptionList(
+        toAddress: toAddress,
+        recommendedFees: recommendedFees,
+      );
     } catch (e) {
       _log.severe("fetchFeeOptions error", e);
       emit(FeeOptionsState(error: extractExceptionMessage(e, getSystemAppLocalizations())));
       rethrow;
     }
   }
 
-  Future<List<FeeOption>> _constructFeeOptionList(
-    String toAddress,
-    RecommendedFees recommendedFees,
-  ) async {
+  Future<List<FeeOption>> _constructFeeOptionList({
+    required String toAddress,
+    required RecommendedFees recommendedFees,
+  }) async {
     final List<FeeOption> feeOptions = [
       FeeOption(
         processingSpeed: ProcessingSpeed.economy,
         waitingTime: const Duration(minutes: 60),
-        fee: await _calculateTransactionFee(toAddress, recommendedFees.hourFee),
+        fee: await _calculateTransactionFee(
+          toAddress: toAddress,
+          satPerVbyte: recommendedFees.hourFee,
+        ),
         feeVByte: recommendedFees.hourFee,
       ),
       FeeOption(
         processingSpeed: ProcessingSpeed.regular,
         waitingTime: const Duration(minutes: 30),
-        fee: await _calculateTransactionFee(toAddress, recommendedFees.halfHourFee),
+        fee: await _calculateTransactionFee(
+          toAddress: toAddress,
+          satPerVbyte: recommendedFees.halfHourFee,
+        ),
         feeVByte: recommendedFees.halfHourFee,
       ),
       FeeOption(
         processingSpeed: ProcessingSpeed.priority,
         waitingTime: const Duration(minutes: 10),
-        fee: await _calculateTransactionFee(toAddress, recommendedFees.fastestFee),
+        fee: await _calculateTransactionFee(
+          toAddress: toAddress,
+          satPerVbyte: recommendedFees.fastestFee,
+        ),
         feeVByte: recommendedFees.fastestFee,
       ),
     ];
     emit(state.copyWith(feeOptions: feeOptions));
     return feeOptions;
   }
 
-  Future<int> _calculateTransactionFee(String toAddress, int satsPerVbyte) async {
+  Future<int> _calculateTransactionFee({
```

Comment:
Rename to `_prepareSweep`, can also make it public, `prepareSweep`, which can come in handy when testing.

Add `req` inside try block in case there's an error building the request object.

Reply:
Added